src/components/device/ha-device-trigger-picker.ts

```diff
@@ -0,0 +1,148 @@
+import "@polymer/paper-input/paper-input";
+import "@polymer/paper-item/paper-item";
+import "@polymer/paper-item/paper-item-body";
+import "@polymer/paper-dropdown-menu/paper-dropdown-menu-light";
+import "@polymer/paper-listbox/paper-listbox";
+import memoizeOne from "memoize-one";
+import {
+  LitElement,
+  TemplateResult,
+  html,
+  css,
+  CSSResult,
+  property,
+} from "lit-element";
+import { HomeAssistant } from "../../types";
+import { fireEvent } from "../../common/dom/fire_event";
+import computeStateName from "../../common/entity/compute_state_name";
+import {
+  DeviceTrigger,
+  fetchDeviceTriggers,
+  triggersEqual,
+} from "../../data/device_automation";
+
+class HaDeviceTriggerPicker extends LitElement {
+  public hass?: HomeAssistant;
+  @property() public label?: string;
+  @property() public deviceId?: string;
+  @property() public triggers: DeviceTrigger[] = [];
+  @property() public trigger?: DeviceTrigger;
+  @property() public presetTrigger?: DeviceTrigger;
+  private noTrigger: DeviceTrigger = {
+    device_id: this.deviceId || "",
+    platform: "device",
+  };
+
+  private _sortedTriggers = memoizeOne((triggers?: DeviceTrigger[]) => {
+    return triggers || [];
+  });
+
+  protected render(): TemplateResult | void {
+    const noTriggers = this._sortedTriggers(this.triggers).length === 0;
+    return html`
+      <paper-dropdown-menu-light .label=${this.label} ?disabled=${noTriggers}>
+        <paper-listbox
+          slot="dropdown-content"
+          .selected=${this._trigger}
+          attr-for-selected="trigger"
+          @iron-select=${this._triggerChanged}
+          id="listbox"
+        >
+          ${noTriggers
+            ? html`
+                <paper-item .trigger=${this.noTrigger}>
+                  No triggers
+                </paper-item>
+              `
+            : ""}
+          ${this._sortedTriggers(this.triggers).map(
+            (trigger) => html`
+              <paper-item .trigger=${trigger}>
+                ${this.hass!.localize(
+                  `ui.panel.config.automation.editor.triggers.type.device.trigger_type.${
+                    trigger.type
+                  }`,
+                  "entity_id",
+                  trigger.entity_id
+                    ? computeStateName(this.hass!.states[trigger.entity_id])
+                    : "",
+                  "event",
+                  trigger.event
+                )}
+              </paper-item>
+            `
+          )}
+        </paper-listbox>
+      </paper-dropdown-menu-light>
+    `;
+  }
+
+  private get _trigger() {
+    return this.trigger;
+  }
+
+  protected updated(changedProps) {
+    super.updated(changedProps);
+
+    // Reset trigger if device-id has changed
+    if (changedProps.has("deviceId")) {
+      this.noTrigger = { device_id: this.deviceId || "", platform: "device" };
+      if (this.deviceId) {
+        fetchDeviceTriggers(this.hass!, this.deviceId).then((trigger) => {
```

Comment:
Extract this into a new async method. `private async _loadDevice() { }` and use `const trigger = await fetchDeviceTriggers(…)`

Reply:
Fixed.